contracts/foundry-tests/oracles/PoolOracle.t.sol

```diff
@@ -0,0 +1,385 @@
+        // Tuesday, 22 March 2022 00:00:00
+        uint256 timestamp = 1647907200;
+        vm.warp(timestamp);
+        address pool = Mocks.mock("IPool");
+
+        vm.record();
+
+        uint256 cumulativeBalancesRatio = 1000;
+        IPool(pool).cumulativeBalancesRatio.mock(cumulativeBalancesRatio);
+        _poolWasUpdatedOnTheSameBlock(pool);
+
+        oracle.update(pool);
+
+        (uint256 ts, uint256 ratio) = oracle.poolObservations(pool, 0);
+        (, bytes32[] memory writes) = vm.accesses(address(oracle));
+        assertEq(ts, timestamp);
+        assertEq(ratio, cumulativeBalancesRatio);
+        // 24 writes for array initialisation + 2 slots written by the actual update
+        assertEq(writes.length, 26);
+
+        for (uint256 i = 1; i < 24; i++) {
+            cumulativeBalancesRatio++;
+            IPool(pool).cumulativeBalancesRatio.mock(cumulativeBalancesRatio);
+            _poolWasUpdatedOnTheSameBlock(pool);
+
+            // Does not update before periodSize
+            skip(59 minutes + 59 seconds);
+            oracle.update(pool);
+            (, writes) = vm.accesses(address(oracle));
+            assertEq(writes.length, 0);
+
+            // Updates when periodSize is reached
+            skip(1 seconds);
+
+            vm.expectEmit(true, true, true, true);
+            emit ObservationRecorded(pool, i, PoolOracle.Observation(block.timestamp, cumulativeBalancesRatio));
+
+            oracle.update(pool);
+            (ts, ratio) = oracle.poolObservations(pool, i);
+            (, writes) = vm.accesses(address(oracle));
+            assertEq(ts, block.timestamp);
+            assertEq(ratio, cumulativeBalancesRatio);
+            assertEq(writes.length, 2);
+
+            // Does not update again within the same periodSize
+            skip(1 seconds);
+            oracle.update(pool);
+            (, writes) = vm.accesses(address(oracle));
+            assertEq(writes.length, 0);
+
+            // Make math simpler for next cycle
+            rewind(1 seconds);
+        }
+
+        // Next update is written on the 0 index
+        skip(1 hours);
+        oracle.update(pool);
+        (ts, ratio) = oracle.poolObservations(pool, 0);
+        (, writes) = vm.accesses(address(oracle));
+        assertEq(ts, block.timestamp);
+        assertEq(ratio, cumulativeBalancesRatio);
+        assertEq(writes.length, 2);
+    }
+
+    function testPeek() public {
+        address pool = Mocks.mock("IPool");
+
+        // Saturday, 26 February 2022 10:15:28
+        vm.warp(1645870528);
+        IPool(pool).cumulativeBalancesRatio.mock(6081248333687398745122493757731327);
+        IPool(pool).getCache.mock(1102997721918618810252394, 1199964191067551228067732, 1645859234);
+        oracle.update(pool);
+
+        vm.record();
+
+        // Sunday, 27 February 2022 10:15:27 (24h after)
+        vm.warp(1645956927);
+
+        PoolOracle.Observation memory oldestObservation = oracle.getOldestObservationInWindow(pool);
```

Comment:
its awesome that we are including the wolframalpha links.  

its still kinda hard to figure out what this test is doing though.  can we add some comments throughout to assist future reviewers.  like you did above in testUpdate